packages/react-native-sdk/src/components/Livestream/ViewerLivestream/ViewerLivestream.tsx

```diff
@@ -92,9 +96,55 @@ export const ViewerLivestream = ({
     },
   };
 
+  const { useIsCallLive } = useCallStateHooks();
+  const isCallLive = useIsCallLive();
+  console.log('🚀 ~ isCallLive:', isCallLive);
+
+  /**
+   * The call is joined using the anonymous user/client.
+   */
+  const handleJoinCall = async () => {
+    try {
+      if (!(call && isCallLive)) {
+        return;
+      }
+      if (
+        [CallingState.JOINED, CallingState.JOINING].includes(
+          call.state.callingState,
+        )
+      ) {
+        setCallJoined(true);
+        return;
+      }
+      await call?.join();
+      setCallJoined(true);
+    } catch (error) {
+      console.error('Failed to join call', error);
+    }
+  };
+
+  const callingState = call?.state.callingState;
```

Comment:
```suggestion
  const callingState = useCallCallingState();
```
Accessing the state `getter` directly isn't reactive and will lead to many bugs. Please use the hook.